### ISSUE TITLE
component attribute transfer

### ISSUE THREAD

**Issue by iij0** [menyifang/ADGAN#5]
How should I work on component attribute transfer? The test.py only evaluates the pose transfer. Could you please provide the code for component attribute transfer?

**Comment by menyifang** [comment]
Hi @iij0, for component attribute transfer, you can give two source person images for style encoding and combine their style codes to get the mixed style code. The style code (len:512) consists of 8 components(len:64): 0Background, 1Pants, 2Hair, 3Skirt, 4Face, 5UpperClothes, 6arm, 7leg.

**Comment by JiamingFB** [comment]
Hi @menyifang ,

For the list "0Background, 1Pants, 2Hair, 3Skirt, 4Face, 5UpperClothes, 6arm, 7leg", is it defined in the code (VggStyleEncoder) or is it a result specific to the pretrained model?

**Comment by menyifang** [comment]
Hi @JiamingFB, yes, the component list is defined by the segmentation labels. With such training data, the model will try to learn a specific attribute in the fixed position, according to the segmentation labels.

**Comment by newExplore-hash** [comment]
> How should I work on component attribute transfer? The test.py only evaluates the pose transfer. Could you please provide the code for component attribute transfer?

yes, i not find any code about this part and i have tried to implement this function( component attribute transfer), But it doesn't seem to work very well，Have you implemented this part of code？

**Comment by artemkurylev** [comment]
> Hi @JiamingFB, yes, the component list is defined by the segmentation labels. With such training data, the model will try to learn a specific attribute in the fixed position, according to the segmentation labels.

Thanks for your explanation. So if I want to do component transfer I will need to retrain the model with changed segmentation labels?

**Comment by NatsumeTetsuya** [comment]
Same as you. Just mix them up can't get suitable result. So does retrain the model work? @artemkurylev 
Are  there some of details of style-code mix? @menyifang

**Comment by fengbuck** [comment]
> Hi @iij0, for component attribute transfer, you can give two source person images for style encoding and combine their style codes to get the mixed style code. The style code (len:512) consists of 8 components(len:64): 0Background, 1Pants, 2Hair, 3Skirt, 4Face, 5UpperClothes, 6arm, 7leg.

I have compeleted to mix the style code. In my way, if I want to exchange upperclothes, I exchange the two images' stylecode[320:383]. Then use decode to generate a new image. But the result is not very well. Is my method correct?
![image](https://user-images.githubusercontent.com/44357986/123911475-6eb7ac80-d9ae-11eb-8bfa-d43d0bafb156.png)

**Comment by abhigoku10** [comment]
@fengbuck @artemkurylev  can you please share the code for attribute transfer ?? THanks in adavance

**Comment by bill20th** [comment]
Nice to meet you @menyifang ,@abhigoku10!

I also need the code for component　attribute　transfer.
Also, where can I find the keypoint_mix in custom_dataset_data_loader.py?
I would appreciate your answer.
